app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts

```diff
@@ -241,6 +241,15 @@ export class AwsServerGroupConfigurationService {
             loadBalancerReloader = this.refreshLoadBalancers(cmd, true);
           }
         }
+
+        if (cmd.targetGroups && cmd.targetGroups.length) {
+          // verify all target groups are accounted for; otherwise, try refreshing load balancers cache
+          const targetGroupNames = this.getTargetGroupNames(cmd);
+          if (intersection(targetGroupNames, cmd.targetGroups).length < cmd.targetGroups.length) {
+            loadBalancerReloader = this.refreshLoadBalancers(cmd, true);
```

Comment:
i guess if you're attached to target groups _and_ classic LBs _and_ some of both lists are missing in the cache, a double refresh could fire off and we'd block until the second one came back. not particularly worried about it 🤣